packages/dds/tree/src/test/tableSchema.spec.ts

```diff
@@ -365,29 +404,350 @@ describe("TableFactory unit tests", () => {
+			);
+		});
 	});
 
 	describe("Delete rows", () => {
-		// Test TODOs:
-		// - Success case: delete empty list (no-op).
-		// - Success case: delete single row.
-		// - Success case: delete multiple rows.
-		// - Failure case: 1 or more specified rows do not exist in the table.
+		it("delete empty list", () => {
+			const { treeView } = createTableTree();
+			treeView.initialize({
+				columns: [],
+				rows: [],
+			});
+
+			treeView.root.deleteAllRows();
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [],
+			});
+		});
+
+		it("delete single row", () => {
+			const { treeView, Row } = createTableTree();
+			const row0 = new Row({ id: "row-0", cells: {} });
+			const row1 = new Row({ id: "row-1", cells: {} });
+			treeView.initialize({
+				columns: [],
+				rows: [row0, row1],
+			});
+
+			// Delete row0
+			treeView.root.deleteRows([row0]);
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [{ id: "row-1", cells: {} }],
+			});
+
+			// Delete row1
+			treeView.root.deleteRows([row1]);
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [],
+			});
+		});
+
+		it("delete multiple rows", () => {
+			const { treeView, Row } = createTableTree();
+			const row0 = new Row({ id: "row-0", cells: {} });
+			const row1 = new Row({ id: "row-1", cells: {} });
+			const row2 = new Row({ id: "row-2", cells: {} });
+			const row3 = new Row({ id: "row-3", cells: {} });
+			treeView.initialize({
+				columns: [],
+				rows: [row0, row1, row2, row3],
+			});
+
+			// Delete rows 1 and 3
+			treeView.root.deleteRows([row1, row3]);
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [
+					{
+						id: "row-0",
+						cells: {},
+					},
+					{
+						id: "row-2",
+						cells: {},
+					},
+				],
+			});
+
+			// Delete rows 0 and 3
+			treeView.root.deleteRows([row0, row2]);
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [],
+			});
+		});
+
+		it("deleting single row that doesn't exist on table fails", () => {
+			const { treeView, Row } = createTableTree();
+			treeView.initialize({
+				columns: [],
+				rows: [],
+			});
+
+			assert.throws(
+				() => treeView.root.deleteRows([new Row({ id: "row-0", cells: {} })]),
+				validateUsageError(/Expected non-negative index, got -1./),
+			);
+		});
+
+		it("deleting multiple rows that doesn't exist on table fails", () => {
+			const { treeView, Row } = createTableTree();
+			treeView.initialize({
+				columns: [],
+				rows: [],
+			});
+
+			assert.throws(
+				() =>
+					treeView.root.deleteRows([
+						new Row({ id: "row-0", cells: {} }),
+						new Row({ id: "row-1", cells: {} }),
+					]),
+				validateUsageError(/Expected non-negative index, got -1./),
+			);
+		});
 	});
 
 	describe("Delete cell", () => {
-		// Test TODOs:
-		// - Success case: valid key with existing data.
-		// - Success case: valid key with no data.
-		// - Failure case: invalid key
+		it("delete valid cell with existing data.", () => {
```

Comment:
These descriptions are no sentences (it doesn't start with a capital letter and isn't grammatically a sentence, its closer to a verb phrase I think), so they shouldn't end with a period.